### ISSUE TITLE
NPE when generationNotMatch fails

### ISSUE THREAD

**Issue by dimo414** [GoogleCloudPlatform/gcloud-java#731]
The following:
```
bucket.create("file", new byte[0], "text/plain");
Blob blob = bucket.get("file");
bucket.storage().create(blob, new byte[0], BlobTargetOption.generationNotMatch());
```
Causes a:
```
com.google.gcloud.storage.StorageException: java.lang.NullPointerException
    at com.google.gcloud.storage.StorageException.translateAndThrow(StorageException.java:74)
    at com.google.gcloud.storage.StorageImpl.create(StorageImpl.java:150)
    at com.google.gcloud.storage.StorageImpl.create(StorageImpl.java:129)
    ...
```

The `translateAndThrow()` call suppresses the underlying stack trace, but this shouldn't trigger an NPE. I do expect it to fail, but it should fail with a meaningful error message.

**Comment by mziccard** [comment]
Hi @dimo414 thanks for the report! This error is due to some information that went missing in the response body. We already fixed this in #725, If you update to the latest version of gcloud-java (0.1.5) you should see this exception for your example:
```
Caused by: com.google.gcloud.storage.StorageException: 304 Not Modified
	at com.google.gcloud.spi.DefaultStorageRpc.translate(DefaultStorageRpc.java:95)
	at com.google.gcloud.spi.DefaultStorageRpc.create(DefaultStorageRpc.java:132)
	at com.google.gcloud.storage.StorageImpl$3.call(StorageImpl.java:145)
	at com.google.gcloud.storage.StorageImpl$3.call(StorageImpl.java:142)
	at com.google.gcloud.RetryHelper.doRetry(RetryHelper.java:181)
	at com.google.gcloud.RetryHelper.runWithRetries(RetryHelper.java:247)
	at com.google.gcloud.RetryHelper.runWithRetries(RetryHelper.java:237)
	at com.google.gcloud.storage.StorageImpl.create(StorageImpl.java:142)
	at com.google.gcloud.storage.StorageImpl.create(StorageImpl.java:129)
	at com.google.gcloud.examples.storage.StorageExample.main(StorageExample.java:548)
```

**Comment by dimo414** [comment]
Verified! Apologies for the stale report.
